### PR TITLE
Expanded Support for AWS Auth Method

### PR DESCRIPTION
**Pull request by stevefranks**
[Proposal]
This PR adds support for authentication via the AWS auth backend. I have tested this code against my local vault instance. I'll add more documentation once we are in agreement on the design.

References: https://github.com/hvac/hvac/issues/479

cc @jeffwecan

### CODE REVIEW

**Comment by jeffwecan** [comment]
Just as a heads-up, I personally haven't had time to look over this PR but intend to do so as soon as I am able.

File: hvac/api/auth_methods/aws.py [hunk 1]
```diff
@@ -0,0 +1,767 @@
...
+        :type cert_name: str | unicode
+        :param mount_point: The "path" the aws auth method was mounted on
+        :return: The data key from the JSON response of the request.
+        :rtype: dict
+        """
+        api_path = '/v1/auth/{0}/config/certificate/{1}'.format(mount_point, cert_name)
+        response = self._adapter.get(
+            url=api_path,
+        )
+        return response.json().get('data')
```
Comment (drewmullen) [inline hvac/api/auth_methods/aws.py]: @jeffwecan we should consider standardizing json returns. personally i like this where the data key is returned directly, however, we have some that doesnt extract it. 

example: https://github.com/hvac/hvac/blob/c013371a43d30b5b54407f3fea2f2fd76bbe5934/hvac/api/auth_methods/ldap.py#L129

File: hvac/api/auth_methods/__init__.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 from hvac.api.auth_methods.mfa import Mfa
 from hvac.api.auth_methods.okta import Okta
 from hvac.api.auth_methods.radius import Radius
+from hvac.api.auth_methods.aws import Aws
```
Comment (drewmullen) [inline hvac/api/auth_methods/__init__.py]: nit picky - the other imports are alphabetical

**Comment by drewmullen** [comment]
LGTM! thanks for the PR! i dont have a strong exp with aws auth so ill let @jeffwecan comment b4 approving

**Comment by jeffwecan** [comment]
Also I realize now that I skipped past the "I'll add more documentation once we are in agreement on the design" point you mentioned when opening the PR. However you're still welcome to fill in those gaps at any point in the future (and it would be much appreciated in fact!). The docstrings will at least cover some things since they end up in "autodoc" output at https://hvac.readthedocs.io/en/stable/source/hvac_api_auth_methods.html.

**Comment by avoidik** [comment]
my apologies for bumping this old PR, but how to use iam_login?

my previous code is

```
vault_client = hvac.Client(url=vault_addr)
vault_client.auth_aws_iam(key_id, secret_id, session_token, vault_iam_header, vault_iam_mount, vault_iam_role)
```

**Comment by avoidik** [comment]
nvm, it just

```
vault_client.auth.aws.iam_login()
```

**Pull request merged**